### PR TITLE
Openai assistant function usage notebook

### PR DESCRIPTION
**Pull request by IANTHEREAL**
## Why are these changes needed?

add a Notebook demonstrates how to leverage OSS Insight (Open Source Software Insight) for advanced GitHub data analysis by **defining function calls in AutoGen for the OpenAI Assistant**.

## Related issue number

https://github.com/microsoft/autogen/issues/602

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: notebook/agentchat_oai_assistant_function_call.ipynb [hunk 1]
```diff
@@ -0,0 +1,223 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```
Comment (qingyun-wu) [inline notebook/agentchat_oai_assistant_function_call.ipynb:6]: Add a title for this notebook? E.g., "Automated Agent Chat with OpenAI Assistants using function call in AutoGen: OSS Insights for Advanced GitHub Data Analysis"

**Review by sonichi — COMMENTED** [review]
Add a link to the notebook in
https://microsoft.github.io/autogen/docs/Examples/AgentChat
and
https://microsoft.github.io/autogen/docs/Use-Cases/agent_chat#diverse-applications-implemented-with-autogen
OK to do it in a separate PR.

File: notebook/agentchat_oai_assistant_function_call.ipynb (hunk 1 above)
Comment (IANTHEREAL) [reply notebook/agentchat_oai_assistant_function_call.ipynb:6]: Done

File: notebook/agentchat_oai_assistant_function_call.ipynb [hunk 2]
```diff
@@ -4,19 +4,40 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture --no-stderr\n",
+    "# %pip install \"pyautogen[teachable]"
```
Comment (sonichi) [inline notebook/agentchat_oai_assistant_function_call.ipynb:32]:
```suggestion
    "# %pip install \"pyautogen~=0.2.0b5"
```

**Review by qingyun-wu — APPROVED** [review]

**Pull request merged**